src/OpenLoco/Map/Tile.cpp

```diff
@@ -105,6 +105,17 @@ SurfaceElement* Tile::surface() const
     return nullptr;
 }
 
+BuildingElement* Tile::building() const
```

Comment:
I don't think this is a great idea. In general you should be looking for all buildings on a tile. (I know we aren't going to have any more than one but I think we should design it to allow that)

Reply:
I would just do it in that function. It's how we do it elsewhere. Other solutions might be to create a visitor. `tile.visitAll<BuildingElement>([](BuildingElement& elBuilding){ ... });`

Reply:
I ended up doing it inside the function.